pkg/resource/generator.go

```diff
@@ -103,7 +103,9 @@ func (g *Generator) syncStorage(cr *regopapi.ImageRegistry, modified *bool) erro
 		return err
 	}
 
-	if driver.StorageChanged(cr, modified) {
+	if len(driver.GetStorageName()) == 0 {
+		runCreate = true
```

Comment:
This is wrong.

since you are looking at spec.Storage, this is always going to be true because spec.Storage.BucketName never gets set, only status.